tests/utils.py

```diff
@@ -12,10 +12,12 @@
 
 
 def retry(operation, times=1, wait=1, error_msg=None, name="default"):
```

Comment:
Starting to think this implementation (it's mine, let me shame myself ok) isn't super readable. Maybe we could build some wrapper for running the operation, akin to the way we do `ok, result := SomeOp()` in Go, or how we use `Result` types with `Err` and `Ok` variants in some languages:

```python
def retry(operation, times=1, wait=1, error_msg=None, name="default"):
    def run_and_assert():
        try:
            return None, operation()
        except AssertionError as exc:
            return str(exc), None

    for attempt in range(times):
        error, result = run_and_assert()
        if error is not None:
            LOGGER.info("failed with...")
        else:
            LOGGER.info("success...")
            return result
    else:
        error_msg = (
            error_msg if error_msg is not None
            else default_error_msg(name, attempts, total)
        )
        pytest.fail(error_msg + ": " + error)
```